gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -203,13 +235,43 @@ public static void movePath(FileSystem srcFs, Path src, FileSystem dstFs, Path d
    */
   public static void copyPath(FileSystem srcFs, Path src, FileSystem dstFs, Path dst, Configuration conf) throws IOException {
 
+    copyPath(srcFs, src, dstFs, dst, false, false, conf);
+  }
+
+  /**
+   * Copies data from a src {@link Path} to a dst {@link Path}.
+   *
+   * <p>
+   *   This method should be used in preference to
+   *   {@link FileUtil#copy(FileSystem, Path, FileSystem, Path, boolean, boolean, Configuration)}, which does not handle
+   *   clean up of incomplete files if there is an error while copying data.
+   * </p>
+   *
+   * <p>
+   *   TODO this method does not handle cleaning up any local files leftover by writing to S3.
+   * </p>
+   *
+   * @param srcFs the source {@link FileSystem} where the src {@link Path} exists
+   * @param src the {@link Path} to copy from the source {@link FileSystem}
+   * @param dstFs the destination {@link FileSystem} where the dst {@link Path} should be created
+   * @param dst the {@link Path} to copy data to
+   */
+  public static void copyPath(FileSystem srcFs, Path src, FileSystem dstFs, Path dst, boolean overwrite,
+                              Configuration conf) throws IOException {
+
+    copyPath(srcFs, src, dstFs, dst, false, overwrite, conf);
+  }
+
+  private static void copyPath(FileSystem srcFs, Path src, FileSystem dstFs, Path dst, boolean deleteSource,
```

Comment:
I was more of the mind that any exposed methods shouldn't combine copy and move together.  I'm less concerned if it is a private method.